File: 26.Remove-Duplicates-from-Sorted-Array.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/description/
#
# algorithms
# Easy (38.18%)
# Total Accepted:    458.2K
# Total Submissions: 1.2M

class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hash_map = {}
        idx = 0        

        for n in nums:
            if n not in hash_map:
                hash_map[n] = True
                nums[idx] = n
                idx += 1
        
        return idx

File: 26.Remove-Duplicates-from-Sorted-Array.java
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
//
// algorithms
// Easy (47.38%)
// Total Accepted:    1.8M
// Total Submissions: 3.8M


class Solution {
    
    public int removeDuplicates(int[] nums) {
        int len = nums.length;
        if (len < 2) {
            return len;
        }
        int res = 1;
        for (int i = 1; i < len; i++) {
            if (nums[i] != nums[i - 1]) {
                nums[res++] = nums[i];
            }
        }
        
        return res;
    }
    
}